pkg/i2gw/providers/gce/ir_converter_test.go

```diff
@@ -1084,3 +959,252 @@ func Test_convertToIR(t *testing.T) {
 func ptrTo[T any](a T) *T {
 	return &a
 }
+
+func getTestIngress(ingressNamespace, ingressName, serviceName string, isExternalIngress bool) *networkingv1.Ingress {
+	iPrefix := networkingv1.PathTypePrefix
+
+	ing := networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingressName,
+			Namespace: ingressNamespace,
+		},
+		Spec: networkingv1.IngressSpec{
+			Rules: []networkingv1.IngressRule{{
+				Host: "test.mydomain.com",
+				IngressRuleValue: networkingv1.IngressRuleValue{
+					HTTP: &networkingv1.HTTPIngressRuleValue{
+						Paths: []networkingv1.HTTPIngressPath{{
+							Path:     "/",
+							PathType: &iPrefix,
+							Backend: networkingv1.IngressBackend{
+								Service: &networkingv1.IngressServiceBackend{
+									Name: serviceName,
+									Port: networkingv1.ServiceBackendPort{
+										Number: 80,
+									},
+								},
+							},
+						}},
+					},
+				},
+			}},
+		},
+	}
+	if isExternalIngress {
+		ing.Annotations = map[string]string{networkingv1beta1.AnnotationIngressClass: gceIngressClass}
+	} else {
+		ing.Annotations = map[string]string{networkingv1beta1.AnnotationIngressClass: gceL7ILBIngressClass}
+	}
+
+	return &ing
+}
+
+func TestGetBackendConfigMapping(t *testing.T) {
+	t.Parallel()
+	testNamespace := "test-namespace"
+
+	testServiceName := "test-service"
+	testBeConfigName1 := "backendconfig-1"
+	testBeConfigName2 := "backendconfig-2"
+	backendConfigs := map[types.NamespacedName]*backendconfigv1.BackendConfig{
+		{Namespace: testNamespace, Name: testBeConfigName1}: {},
+		{Namespace: testNamespace, Name: testBeConfigName2}: {},
+	}
+	expectedServices := serviceNames{
+		{Namespace: testNamespace, Name: testServiceName},
+	}
+
+	testCases := []struct {
+		desc             string
+		services         map[types.NamespacedName]*apiv1.Service
+		expectedServices serviceNames
+	}{
+		{
+			desc: "Specify BackendConfig with cloud.google.com/backend-config annotation, using the same BackendConfig for all ports",
+			services: map[types.NamespacedName]*apiv1.Service{
+				{Namespace: testNamespace, Name: testServiceName}: {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      testServiceName,
+						Namespace: testNamespace,
+						Annotations: map[string]string{
+							backendConfigKey: `{"default":"backendconfig-1"}`,
+						},
+					},
+				},
+			},
+			expectedServices: expectedServices,
```

Comment:
Why do all test cases have the same expected Services? What is this test actually testing?

Reply:
It is testing: the different ways of using BackendConfig annotation on a Service, so the service doesn't need to be changed.

Reply:
Generally with test cases like this I'd expect an input and expected output var. If this expected output var is not useful, what would be?

Reply:
Yes that makes sense, just updated the test case.